lonestar/matrixcompletion/CMakeLists.txt

```diff
@@ -1,13 +1,18 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ffast-math")
+add_executable(matrixCompletion matrixCompletion.cpp)
+add_dependencies(apps matrixCompletion)
+target_link_libraries(matrixCompletion PRIVATE Galois::shmem lonestar)
+install(TARGETS matrixCompletion DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT apps EXCLUDE_FROM_ALL)
+
+if(CMAKE_COMPILER_IS_GNUCC)
+  target_compile_options(matrixCompletion PRIVATE -ffast-math)
+endif()
 
 find_package(Eigen)
 if(Eigen_FOUND)
-  include_directories(${Eigen_INCLUDE_DIRS})
-  add_definitions(-DHAS_EIGEN -DEIGEN_DONT_PARALLELIZE)
+  target_link_libraries(matrixCompletion Eigen3::Eigen)
```

Comment:
Eigen is header only, so we don't need to worry about linking it.

Reply:
Actually I think the right check for eigen here is:

```
find_package(Eigen3)
if(EIGEN3_FOUND)
  target_compile_definitions(matrixCompletion PRIVATE -DHAS_EIGEN -DEIGEN_DONT_PARALLELIZE)
  include_directories("${EIGEN3_INCLUDE_DIR}")
  MESSAGE("Eigen include dir: ${EIGEN3_INCLUDE_DIR}")
endif()
```

Hopefully the CI systems agree...

Reply:
I thought one of the benefits of the target-based approach was that header-only libraries can be treated uniformly: https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries

Though you are right that our internal FindEigen.cmake module doesn't actually export the right symbols here, but I think I'd rather just fix the module than change the method of importing.